jupyter_book/commands/__init__.py

```diff
@@ -57,218 +59,101 @@ def main():
     help="Which builder to use.",
     type=click.Choice(list(BUILDER_OPTS.keys())),
 )
-def build(path_book, path_output, config, toc, warningiserror, builder):
-    """Convert your book's content to HTML or a PDF."""
-    # Paths for our notebooks
-    PATH_BOOK = Path(path_book).absolute()
-
-    # `book_config` is manual over-rides, `config` is the path to a _config.yml file
-    book_config = {}
-
-    # Table of contents
-    # TODO Set TOC dynamically to default value and let Click handle this check
-    if toc is None:
-        toc = PATH_BOOK.joinpath("_toc.yml")
+def build(path_source, path_output, config, toc, warningiserror, builder):
+    """Convert your book's or page's content to HTML or a PDF."""
+
+    # Paths for the notebooks
+    PATH_SRC_FOLDER = Path(path_source).absolute()
+
+    config_overrides = {}
+    freshenv = False
+    BUILD_PATH = (
+        path_output if path_output is not None else find_config_path(PATH_SRC_FOLDER)
```

Comment:
@AakashGfude so the assumption here is that if path_output is `none` then it calls a `utility` function to search for the `config` file and assume the `_build` folder is in that directory. 

Does `path_output` come from the `default` values in `config` from `jupyter_book`? It looks like the old code had

```
BUILD_PATH = path_output if path_output is not None else PATH_BOOK
```

and I assume `path_book` would be specified from the `CLI` by a user.

Reply:
@mmcky path_output is something which the user can explicitly specify in the command line while building , and `PATH_BOOK` is the source folder for the build

Reply:
https://github.com/executablebooks/jupyter-book/blob/master/jupyter_book/commands/__init__.py#L63  . like here in the master branch, it just took the path of `path_book` which is the source folder

Reply:
Right -- so won't they be the `default` values for the `_build` folder should it exist.  I guess I don't understand why we don't keep using those default paths for setting the `BUILD_PATH`?

Reply:
righ @mmcky , that should be the default value for `_build` in case of `book` but, in case of nested `page`  we want it to be in the parent folders, where the `_build` of `book` would have resided.

Reply:
@mmcky do you want me to move this line inside the `if` statement of `page` only, and put the previously used line there again?. I understand the confusion.

Reply:
Isn't `_page` just a subfolder of the main `_build` folder such as `_build/_page/` or are you building a `_build` and a `_page` at the same level?

Reply:
but the basic functionality of that function is that it returns the `source_path` unless there is any `_config.yml` in the parent folders, in which case it will return the path to that folder. 

So, in the case of `book` it seems full proof to me. 
But in the case of `page` if the parent `book` folder does not have a `_config.yml` file, then the function won't be able to determine the correct folder to create the `_build` for `page` (i,e, the folder where the `_build` for `book` is), and it will default to the path of the `page` only, as the path of `_build`.

For this case, I think the code needs a lot of rearranging and time, so we can create an issue and I can start rearranging code later. But for now, we can merge this PR?

Reply:
ok thanks for explaining. Happy to raise an issue and loop back to this later if needed. I haven't been able to test locally -- but @AakashGfude has done some testing. So happy for this to be merged.